lib/elixir/src/elixir_erl_for.erl

```diff
@@ -146,9 +146,7 @@ build_reduce(Clauses, Expr, {nil, Ann} = Into, Acc, S) ->
   elixir_erl:remote(Ann, lists, reverse,
     [build_reduce_clause(Clauses, ListExpr, Into, Acc, S)]);
 build_reduce(Clauses, Expr, {bin, _, _} = Into, Acc, S) ->
-  {bin, Ann, Elements} = Expr,
-  BinExpr = {bin, Ann, [{bin_element, Ann, Acc, default, [bitstring]} | Elements]},
-  build_reduce_clause(Clauses, BinExpr, Into, Acc, S).
```

Comment:
The old code made the tests fail on compilation. 

Since we changed `comprehension_expr/2`, when we have `bin` into's it's not guaranteed we have a `bin` expression, that's why I've removed this code.

Now it's failing on execution, which I'm investigating now. 

PS.: I know the code is wrong for sure, just committed and opened the PR for reviewing contributions!

Reply:
@kelvinst this means we would need to add the two clauses we just removed back here. So I don't think the change I proposed is worth it.

I think we can drop this particular enhancement. Thank you for investigating! :heart: